app/javascript/dashboard/routes/dashboard/settings/auditlogs/Index.vue

```diff
@@ -88,6 +92,34 @@ export default {
     this.$store.dispatch('auditlogs/fetch', { page: 1 });
   },
   methods: {
+    generateLogText(auditLogItem) {
+      const username =
+        auditLogItem.username !== null
+          ? auditLogItem.username
+          : this.$t('AUDIT_LOGS.ACTION.SYSTEM');
+      const auditableType = auditLogItem.auditable_type.toLowerCase();
+      const action = auditLogItem.action.toLowerCase();
+
+      if (auditLogItem.action === 'create') {
+        return `${username} ${this.$t(
```

Comment:
As per the spec, the name should have the preference. Resolve the name from the agent model in the vuex. If it is not available, then use the email as the fallback.

Reply:
done